impl/implqueue/pubsub.go

```diff
@@ -103,3 +109,11 @@ func (p *pubsubConsumer) Close() error {
 	})
 	return nil
 }
+
+func getMetadataFromMessage(msg *pubsub.Message) map[string][]byte {
```

Comment:
What do you think of adding `if len(msg.Attributes) == 0 {return nil}` on the beginning of this func? That way you don't have to pass forward an initialized empty map.

Reply:
I'm afraid that by doing so, I'm not always sure that `attr[key]` runs without panic...
I know it's bad to keep creating maps that way, but it is annoying to always check `if attr == nil` before using it.

Reply:
Yes, but then you could not expose the map and instead of `Metadata() map[string][]byte` you could have `Metadata(key string) []byte`. This way you also have the advantage of an immutable metadada collection.